behaviors/v1_1/management-chain.md

```diff
@@ -0,0 +1,237 @@
+    * Maintains a single cache entry per `block.timestamp` (reference time -> offChain data). This implies the same context for the entire block.
+    * Management-VC availability failures :
+        * Available outdated information - Consumer-VC "partially halts": application state does not progress, only system state can be updated (elected validators set, reputation..).
+        * Unavailable - currently treated the same as above.
+    * Elected validators provider := [_ElectedValidators](../smart-contracts/system/_ElectedValidators.md) system contract. 
+        * The Reference Elected Validators is fetched from the Management-VC by querying the `_ElectedValidators` contract.
+    
+* Private-VC
+    * Used for example for testing.
+    * Does not depend on another VC to work.
+    * Hierarchical VCs design can still hold where other (Private) Consumer-VCs use it as their Management-VC.
+    * Elected validators provider := [_ElectedValidators](../smart-contracts/system/_ElectedValidators.md) system contract in "dummy" mode - with rudimentary logic. 
+    
+* Cross VCs communication:
+    * Intra Validator node (across its VC instances ) - Direct HTTP Synchronous rpc calls.
+
+* Audit node:
+    * Audit node implementation relies on historical VC state queries (by block height).
+  
+* Node - block sync:
+    * Not affected. Note that in non-audit mode.
+  
+* Management-VC protocol upgrade:
+    * CrossChainAPI should be backward compatible.
+
+* Cross VC queries:
+    * First a reference chain block height is queried by time.
+      * Latest committed block prior to the provided time.
+      * The query is non-deterministic with a +/-1 variation and is treated as off-chain data. (similar to block time stamp).
+    * Data is queried by height (deterministic).
+  
+* Elected validators configuration:
+    * Expired elected validators set - allows partial state progress.
+        * Only "system state" update: elected set, reputation, no application transactions.
+    * Transitioning to a new configuration is done agreement by writing to state. The configuration holds during the entire consensus round (updating state at round i determines the config of round i+1).
+
+## Virtual Chain Flows:
+
+### RequestOrderedCommittee flow (Read)
+* At the beginning of each consensus round (block height), the node queries the `ConsensusContext` for a sorted list of nodes (committee).
+* The `ConsensusContext` retrieves the current Elected Validators Set from state.
+    * Poll until success (retry on error).
+        * Get the Elected Validators state record  by calling `_ElectedValidators.getElectedValidators()` (using the `VM.CallSystemContract`).
+        * If the Elected Validators state record is empty (state update has yet to occur) the `ConsensusContext`  the `_ElectedValidators` will return the Reference Validator Set by calling `_Provision.getReferenceElectedValidators()` (currently not supported).
```

Comment:
I don't understand this statement. The record is only empty for the first time ever. Otherwise, it holds the state from the previous block. What am I missing?
Also, `_Provision` - currently not supported - then why is it in the spec? :)

Reply:
imho, there are two situations where the state record may be empty: 
1. the VC was launched during a management chain outage - in this case blocks will be closed on the VC in "limited" availability mode: only tx involving polling of the management chain for elected validators may be included in a block
1. "Private" VC

@gadcl any other scenarios?